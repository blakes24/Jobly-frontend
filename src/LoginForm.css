.LoginForm .card-title {
  font-size: 2rem;
  text-align: center;
}

.LoginForm .card {
  -webkit-box-shadow: 5px 5px 10px 2px lightgrey;
  box-shadow: 3px 3px 10px 2px lightgrey;
  max-width: 500px;
  margin: 0 auto;
  margin-top: 15vh;
}

.LoginForm form {
  padding: 1rem;
}

.LoginForm label {
  margin: 10px 0 5px;
}

.LoginForm .btn {
  margin-top: 1rem;
}

@media screen and (max-width: 600px) {
  .LoginForm .card {
    margin-top: 1rem;
    box-shadow: none;
    border: none;
  }

  .LoginForm .card-body {
    background-color: #f1f2f3;
    font-weight: 500;
  }
}
