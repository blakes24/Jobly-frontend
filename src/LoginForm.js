import React from 'react';
import { useFormik } from 'formik';
import { Form, Button } from 'react-bootstrap';
import { useHistory } from 'react-router-dom';

const validate = (values) => {
  const errors = {};
  if (!values.username) {
    errors.username = 'Required';
  }
  if (!values.password) {
    errors.password = 'Required';
  }
  return errors;
};

const LoginForm = ({ login }) => {
  const history = useHistory();
  const formik = useFormik({
    initialValues : {
      username : '',
      password : ''
    },
    validate,
    onSubmit      : async (values) => {
      await login(values);
      history.push('/');
    }
  });

  return (
    <Form onSubmit={formik.handleSubmit}>
      <Form.Label htmlFor="username">Username</Form.Label>
      <Form.Control
        id="username"
        name="username"
        type="text"
        onChange={formik.handleChange}
        onBlur={formik.handleBlur}
        value={formik.values.username}
      />
      {formik.touched.username && formik.errors.username ? (
        <Form.Text className="text-danger">{formik.errors.username}</Form.Text>
      ) : null}

      <Form.Label htmlFor="password">Password</Form.Label>
      <Form.Control
        id="password"
        name="password"
        type="password"
        onChange={formik.handleChange}
        onBlur={formik.handleBlur}
        value={formik.values.password}
      />
      {formik.touched.password && formik.errors.password ? (
        <Form.Text className="text-danger">{formik.errors.password}</Form.Text>
      ) : null}

      <Button type="submit">Submit</Button>
    </Form>
  );
};

export default LoginForm;
