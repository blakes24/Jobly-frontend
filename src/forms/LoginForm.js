import React from 'react';
import { useFormik } from 'formik';
import { Form, Button, Card } from 'react-bootstrap';
import { useHistory } from 'react-router-dom';
import './LoginForm.css';

const validate = (values) => {
  const errors = {};
  if (!values.username) {
    errors.username = 'Required';
  }
  if (!values.password) {
    errors.password = 'Required';
  }
  return errors;
};

const LoginForm = ({ login }) => {
  const history = useHistory();
  const formik = useFormik({
    initialValues : {
      username : '',
      password : ''
    },
    validate,
    onSubmit      : async (values) => {
      try {
        await login(values);
        history.push('/');
      } catch (err) {
        formik.errors.password = err;
      }
    }
  });

  return (
    <div className="LoginForm">
      <Card className="LoginForm-card">
        <Card.Body>
          <Card.Title>Log In</Card.Title>
          <Form onSubmit={formik.handleSubmit}>
            <Form.Label htmlFor="username">Username</Form.Label>
            <Form.Control
              id="username"
              name="username"
              type="text"
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              value={formik.values.username}
            />
            {formik.touched.username && formik.errors.username ? (
              <Form.Text className="text-danger">{formik.errors.username}</Form.Text>
            ) : null}

            <Form.Label htmlFor="password">Password</Form.Label>
            <Form.Control
              id="password"
              name="password"
              type="password"
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              value={formik.values.password}
            />
            {formik.touched.password && formik.errors.password ? (
              <Form.Text className="text-danger">{formik.errors.password}</Form.Text>
            ) : null}

            <Button type="submit">Log In</Button>
          </Form>
        </Card.Body>
      </Card>
    </div>
  );
};

export default LoginForm;
