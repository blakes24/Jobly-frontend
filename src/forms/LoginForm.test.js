import { render } from '@testing-library/react';
import LoginForm from './LoginForm';

test('renders without crashing', () => {
  render(<LoginForm />);
});

it('matches snapshot', function() {
  const { asFragment } = render(<LoginForm />);
  expect(asFragment()).toMatchSnapshot();
});
