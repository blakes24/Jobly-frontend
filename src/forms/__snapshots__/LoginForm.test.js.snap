// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="LoginForm"
  >
    <div
      class="LoginForm-card card"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          Log In
        </div>
        <form
          class=""
        >
          <label
            class="form-label"
            for="username"
          >
            Username
          </label>
          <input
            class="form-control"
            id="username"
            name="username"
            type="text"
            value=""
          />
          <label
            class="form-label"
            for="password"
          >
            Password
          </label>
          <input
            class="form-control"
            id="password"
            name="password"
            type="password"
            value=""
          />
          <button
            class="btn btn-primary"
            type="submit"
          >
            Log In
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
